c10/cuda/CUDADeviceAssertionHost.h

```diff
@@ -7,9 +7,7 @@
 #include <string>
 #include <vector>
 
-#ifdef USE_CUDA
 #define TORCH_USE_CUDA_DSA
```

Comment:
we don't even need the #ifdefs if it's always defined, we could just clean up all #ifdefs. @r-barnes does it make sense?

Alternatively, we can turn it on by default in CMake, but that'd allow to turn it off if needed.